astropy/units/typing.py

```diff
@@ -0,0 +1,66 @@
+"""Typing module for supporting type annotations related to :mod:`~astropy.units`."""
+
+from __future__ import annotations
+
+__all__ = ["QuantityLike"]
+
+
+from typing import TYPE_CHECKING, Union
+
+import numpy.typing as npt
+
+from astropy.units import Quantity
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+# Note: Quantity is technically covered by npt.ArrayLike, but we want to
+# explicitly include it here so that it is clear that we are also including
+# Quantity objects in the definition of QuantityLike.
+QuantityLike: TypeAlias = Union[Quantity, npt.ArrayLike]
+"""Type alias for a quantity-like object.
+
+This is an object that can be converted to a :class:`~astropy.units.Quantity` object
+using the :func:`~astropy.units.Quantity` constructor.
+
+Examples
+--------
+We assume the following imports:
+
+    >>> from astropy import units as u
+
+This is a non-exhaustive list of examples of quantity-like objects:
+
+Integers and floats:
+
+    >>> u.Quantity(1, u.meter)
+    <Quantity 1.0 m>
+
+    >>> u.Quantity(1.0, u.meter)
+    <Quantity 1.0 m>
+
+Tuples and lists:
+
+    >>> u.Quantity((1.0, 2.0), u.meter)
```

Comment:
I always found it a bit of a pity that numpy allowed tuple other than for structured arrays; we are somewhat stuck with it, and here it is probably more important to reflect what the code does rather than what I wish it did, but let's at least put list first!

Reply:
Done!